dl1_data_handler/image_mapper.py

```diff
@@ -1,744 +1,1036 @@
+            ],
+            axis=-1,
+        )
+        return images
 
-            if map_method in [
-                "rebinning",
-                "nearest_interpolation",
-                "bilinear_interpolation",
-                "bicubic_interpolation",
-            ]:
-                self.image_shapes[camtype] = self.interpolation_image_shape[camtype]
-
-            # At the edges of the cameras the mapping methods run into issues.
-            # Therefore, we are using a default padding to ensure that the camera pixels aren't affected.
-            # The default padding is removed after the conversion is finished.
-            if map_method in ["image_shifting", "axial_addressing", "indexed_conv"]:
-                self.default_pad = 0
-            elif map_method == "bicubic_interpolation":
-                self.default_pad = 3
-            else:
-                self.default_pad = 2
-
-            if map_method != "oversampling" or camtype in [
-                "ASTRICam",
-                "CHEC",
-                "SCTCam",
-            ]:
-                self.image_shapes[camtype] = (
-                    self.image_shapes[camtype][0] + self.default_pad * 2,
-                    self.image_shapes[camtype][1] + self.default_pad * 2,
-                    self.image_shapes[camtype][2],
-                )
-            else:
-                self.image_shapes[camtype] = (
-                    self.image_shapes[camtype][0] + self.default_pad * 4,
-                    self.image_shapes[camtype][1] + self.default_pad * 4,
-                    self.image_shapes[camtype][2],
-                )
+    def _get_virtual_pixels(self, x_ticks, y_ticks, pix_x, pix_y):
+        """Get the virtual pixels outside of the camera."""
+        gridpoints = np.array(np.meshgrid(x_ticks, y_ticks)).T.reshape(-1, 2)
+        gridpoints = [tuple(l) for l in gridpoints.tolist()]
+        virtual_pixels = set(gridpoints) - set(zip(pix_x, pix_y))
+        virtual_pixels = np.array(list(virtual_pixels))
+        return virtual_pixels
 
-            # Initializing the indexed matrix
-            self.index_matrixes[camtype] = None
-            # Calculating the mapping tables for the selected camera types
-            self.mapping_tables[camtype] = self.generate_table(camtype)
+    def _create_virtual_hex_pixels(
+        self, first_ticks, second_ticks, first_pos, second_pos
+    ):
+        """Create virtual hexagonal pixels outside of the camera."""
+        dist_first = np.around(abs(first_ticks[0] - first_ticks[1]), decimals=3)
+        dist_second = np.around(abs(second_ticks[0] - second_ticks[1]), decimals=3)
+
+        tick_diff = len(first_ticks) * 2 - len(second_ticks)
+        tick_diff_each_side = tick_diff // 2
+        # Extend second_ticks
+        for _ in range(tick_diff_each_side + self.internal_pad * 2):
+            second_ticks = (
+                [np.around(second_ticks[0] - dist_second, decimals=3)]
```

Comment:
Suggestion. Add decimals precision as a constant in order to remove MAGIC NUMBERS. 
Example: 

```
from collections import namedtuple
Constants = namedtuple('Constants', ['decimal_precision', 'tick_interval_limit'])
constants = Constants(3, 0.002)
print(constants.decimal_precision)
```